### ISSUE TITLE
Disable replacing quotes

### ISSUE THREAD

**Issue by luisdavim** [redhat-developer/yaml-language-server#246]
In yaml single and double quotes have different meanings (https://yaml-multiline.info/) and I want to be able to auto format my file but keep the quotes as they are.
I see an option to choose between single or double quotes but no option to just keep them as they are.

**Comment by ok-nick** [comment]
Is it a bug that quotes are being replaced? I'd like `yaml-language-server` to strictly format my files but it doesn't seem to enforce quotes, even though I have `yaml.format.singleQuote` set to `false`. Does this setting only pertain to certain circumstances?

**Comment by luisdavim** [comment]
Yes, it is a but that quotes are replaced because the different quote types have different meanings.

**Comment by donfiguerres** [comment]
I guess it's a feature enhancement that has to be done in the `prettier` project because the language server simply passes the option to the `prettier` `format` function.

https://github.com/redhat-developer/yaml-language-server/blob/cf3f79202524ba979f1f4011636b85d08d81bdcb/src/languageservice/services/yamlFormatter.ts#L40

**Comment by luisdavim** [comment]
Thanks, I've found a related issue there and left a comment https://github.com/prettier/prettier/issues/5350#issuecomment-1513743949
